app/controllers/secrets_controller.rb

```diff
@@ -92,12 +92,10 @@ def get_secret_from_variable(variable)
     end
   end
 
-  def audit_fetch resource, version: nil
-    # don't audit the fetch if the resource doesn't exist
-    return unless resource
-
+  def audit_fetch (resource, version: nil)
     fetch_info = error_info.merge(
       resource: resource,
+      resource_id: resource ? nil : resource_id,
```

Comment:
Please consider using only `resource_id` without `resource`. 
If `resource` exists, take its id. otherwise, use `resource_id`

Reply:
I tried to do this exact  thing but for some reason when resource existed and resource.id was sent to fetch.rb it was empty... I dont really know why it happened but resource.id had a value only when the whole resource object was sent and the id property was accessed in fetch.rb. I couldnt find any other way around it except having both resource and resource_id sent.   @orenbm I would like to hear what you think about that too